app/controllers/public_health_controller.rb

```diff
@@ -100,7 +100,11 @@ def tab_counts
     # Get patients by workflow and tab
     patients = patients_by_type(workflow, tab)
 
-    render json: { total: patients.size }
+    # Cache tab sizes
+    cache_key = "#{current_user.jurisdiction.id}-#{workflow}-#{tab}"
+    total = Rails.cache.fetch(cache_key, expires_in: 5.minutes, race_condition_ttl: 30.seconds) { patients.size }
+
+    render json: { total: total }
```

Comment:
It would be good to include `patients_by_type` within the cache miss block instead of so the SQL is not getting run every time. This current code is still going to run `patients_by_type` each time.

Reply:
Does `patients_by_type` actually execute the SQL?

Reply:
Yes. Here is the log with the current version of the code:

```ruby
    # Get patients by workflow and tab
    puts 'Before patients_by_type'
    patients = patients_by_type(workflow, tab)
    puts 'After patients_by_type'

    render json: { total: patients.size }
```

```
Before patients_by_type
  Jurisdiction Load (0.4ms)  SELECT `jurisdictions`.* FROM `jurisdictions` WHERE `jurisdictions`.`id` = 1 LIMIT 1
  ↳ app/controllers/public_health_controller.rb:128:in `patients_by_type'
   (0.6ms)  SELECT `jurisdictions`.`id` FROM `jurisdictions` WHERE ((`jurisdictions`.`ancestry` LIKE '1/%' OR `jurisdictions`.`ancestry` = '1') OR `jurisdictions`.`id` = 1)
  ↳ app/models/jurisdiction.rb:41:in `transferred_in_patients'
  CACHE  (0.0ms)  SELECT `jurisdictions`.`id` FROM `jurisdictions` WHERE ((`jurisdictions`.`ancestry` LIKE '1/%' OR `jurisdictions`.`ancestry` = '1') OR `jurisdictions`.`id` = 1)
  ↳ app/models/jurisdiction.rb:42:in `transferred_in_patients'
   (0.8ms)  SELECT `transfers`.`patient_id` FROM `transfers` WHERE `transfers`.`to_jurisdiction_id` IN (1, 2, 5, 3, 4, 6, 7, 1) AND `transfers`.`from_jurisdiction_id` NOT IN (1, 2, 5, 3, 4, 6, 7, 1) AND (created_at > '2020-08-27 13:44:32.063957')
  ↳ app/models/jurisdiction.rb:43:in `transferred_in_patients'
  CACHE  (0.0ms)  SELECT `jurisdictions`.`id` FROM `jurisdictions` WHERE ((`jurisdictions`.`ancestry` LIKE '1/%' OR `jurisdictions`.`ancestry` = '1') OR `jurisdictions`.`id` = 1)
  ↳ app/models/jurisdiction.rb:44:in `transferred_in_patients'
After patients_by_type
```

And it calls one final count afterwards for `patients.size` which is what this would currently cache.

Reply:
Ah, `transferred_in_patients` and `transferred_out_patients` are doing this. That definitely should be changed. Let me give it some thought and I will do a refactor.